src/common/Requirements.ts

```diff
@@ -116,12 +116,31 @@ const messages = Messages.loadMessages(
 );
 
 export class RequirementProcessor {
+    private static instance: RequirementProcessor;
+    private skipRequirementList: { [key: string]: boolean } = {};
+    private constructor() {}
+
+    public static getInstance() {
+        return (
+            RequirementProcessor.instance ||
+            (RequirementProcessor.instance = new RequirementProcessor())
+        );
+    }
+
+    public resetSkipRequirements() {
+        this.skipRequirementList = {};
+    }
+
+    public skipRequirements(requirementNames: string[]) {
+        requirementNames.forEach((value) => {
+            this.skipRequirementList[value] = true;
+        });
+    }
+
```

Comment:
This change I think will get lots of feedback from both of you 😄  

I'm making `RequirementProcessor` a singleton here so that it can take strings of names of requirements to skip in `execute`.

There are number of reasons for doing this. First, the dot notation to a requirement in `CommandRequirement` to override the `enabled`-ness of `RequirementList` uses dynamic nature of Javascript and does not guarantee of the existence of such requirement. So specifying by the name, in string, forces the caller to know what she is overriding. But if still the name doesn't exist it allows guard rail not to access non-existent property.

Second, `validateInputParameter` method in each of our commands is called inside `run`. This means that the validation doesn't happen until the last minute and sometimes the necessary values aren't set yet(for example,`Preview` command). So providing requirement override in `RequirementProcessor` allows validation to run, have all requirements set, and then be able to decide which ones to skip.

I think this change to `RequirementProcessor` may not be necessary if we can run `validateInputParameter` in each command in their respective constructors. But I'm not sure if it's a good implementation pattern to do so. Also, this still will require how we can convey what named `RequirementList` there are inside `CommandRequirements`.

Reply:
Please take a look at my suggestion about using `validate` function for each flag. That approach might greatly simplify things.

Reply:
`RequirementProcessor` singleton no more :)